kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java

```diff
@@ -481,8 +534,42 @@ private void enableInterface(String deviceId, NetworkProperties properties, Devi
         }
 
     }
+    
+    private void createVirtualInterface(String deviceId, NetworkProperties properties, NMDeviceType deviceType)
+        throws DBusException {
+        Map<String, Map<String, Variant<?>>> newConnectionSettings = NMSettingsConverter.buildSettings(properties,
+                Optional.empty(), deviceId, deviceId, deviceType);
+        DeviceCreationLock dcLock = new DeviceCreationLock(this, deviceId);
+        Settings settings = this.dbusConnection.getRemoteObject(NM_BUS_NAME, NM_SETTINGS_BUS_PATH, Settings.class);
+        DBusPath createdConnectionPath = settings.AddConnection(newConnectionSettings);
+        Connection createdConnection = this.dbusConnection.getRemoteObject(NM_BUS_NAME,
+                createdConnectionPath.getPath(), Connection.class);
+        try {
+            Optional<Device> returnedDevice = dcLock.waitForDeviceCreation();
+            if (!returnedDevice.isPresent()) {
+                logger.warn("Could not obtain device for {}", deviceId);
+                return;
+            }
+            Device createdDevice = returnedDevice.get();
+            if (Boolean.FALSE.equals(this.networkManager.isDeviceManaged(createdDevice))) {
+                this.networkManager.setDeviceManaged(createdDevice, true);
+            }
+            DeviceStateLock dsLock = new DeviceStateLock(this.dbusConnection, createdDevice.getObjectPath(),
+                    NMDeviceState.NM_DEVICE_STATE_ACTIVATED);
+            this.networkManager.activateConnection(createdConnection, createdDevice);
+            dsLock.waitForSignal();
+        } catch (DBusException | TimeoutException e) {
```

Comment:
For debugging reason I would add also `DBusExecutionException` here.

When something breaks at runtime, the underlying library usually throws a `DBusExecutionException` instead of a `DBusException`.

```suggestion
        } catch (DBusExecutionException | DBusException | TimeoutException e) {
```